apps/sms/services/js/activity/activity_shim.js

```diff
@@ -78,31 +42,83 @@
    */
   var ActivityShim = {
     /**
+     * Reference to currently active activity request.
+     * @type {ActivityRequestHandler}
+     */
+    [priv.request]: null,
+
+    /**
      * Initialized activity service bridge.
-     * @type {number} appInstanceId Unique identifier of the app instance that
-     * is used to establish 1-to-1 only connection between this service and
-     * corresponding client hosted in the same app instance.
      */
-    init(appInstanceId) {
-      if (!appInstanceId) {
-        throw new Error('App instance id is not specified!');
+    init() {
+      this.initService();
+
+      // We use setTimeout here to allow activity client to do connection
+      // handshake with the service first, so that it will be able to receive
+      // event broadcasted by service.
+      setTimeout(() => {
```

Comment:
> This can be easily resolved if BridgeServiceMixin would allow to subscribe to bridge service "connect" event and once client is connected we can safely set system message handler. What do you think?

If we have such an event we can try using it. But not in this patch likely ?